homeassistant/components/vallox/__init__.py

```diff
@@ -86,6 +87,19 @@ class ServiceMethodDetails(NamedTuple):
         method="async_set_profile_fan_speed_boost",
         schema=SERVICE_SCHEMA_SET_PROFILE_FAN_SPEED,
     ),
+    SERVICE_SET_PROFILE: ServiceMethodDetails(
+        method="async_set_profile",
+        schema=vol.Schema(
+            {
+                vol.Required("profile"): vol.All(
+                    vol.Coerce(int), vol.Clamp(min=1, max=5)
```

Comment:
Don't use ints as profile. Pass in a string that you would map to a profile.

Reply:
any ideas why originally the `Extra` profile was not 'settable'? It's not included in `VALLOX_PROFILE_TO_PRESET_MODE_SETTABLE`, but I could add it there and cut down on the definitions in `const.py` and use it to map the param.

Reply:
It is some firmware limitation. Vallox's Web UI does not allow to set Extra profile in 2.0.16 (which I currently use).

Reply:
does the HA side support some sort of conditionals based on the firmware version, it works fine on my .24 version

if someone can verify how it works on .20 i could branch it to either >=.20 or >=.24

Reply:
Why do you need that Extra profile? Isn’t fireplace and boost sufficient for all the use cases?

Reply:
It's a separate configurable profile that can be used via digital inputs on the board or via modbus or the websocket api.

I'd use Fireplace for the fireplace, and a differently configured extra profile to compensate for my kitchen air extraction.

![image](https://github.com/home-assistant/core/assets/11215688/96d49483-b510-4728-8c87-c02c0abeaa74)

This way i can avoid pulling a low voltage cable from the kitchen to the air handler, and just trigger the extra profile via a sensor on the kitchen range hood open/close relay.

Reply:
yes, if i apply it, the ui says 

![image](https://github.com/home-assistant/core/assets/11215688/0a5799dc-b700-4457-8587-0318721b2725)